src/main.rs

```diff
@@ -147,81 +153,152 @@ fn run_command(args: &Args) -> Result<(), Error> {
+                addons.push(entry.path());
+            }
         }
-        if args.flag_release {
-            let version = match &p.version {
-                Some(v) => v,
-                None => panic!("Unable to determine version number"),
-            };
-            if args.flag_force {
-                files::clear_release(&version).unwrap();
-                files::clear_pbos(&p).unwrap();
+        if args.arg_addons != "" {
+            let specified_addons: Vec<String> = args.arg_addons.split(",").map(|s| s.to_string()).collect();
+            let addon_path = PathBuf::from("addons");
+            for addon in specified_addons {
+                let mut adn = addon_path.clone();
+                adn.push(&addon);
+                if !adn.exists() {
+                    return Err(error!("addons/{} was not found", addon.bold()));
+                }
+                if skip.contains(&addon) { continue };
+                addons.push(adn);
             }
-            build::release(&p, &version).print_error(true);
-            println!("  {} {} v{}", "Finished".green().bold(), &p.name, version);
         } else {
-            if args.arg_addons != "" {
-                let addons: Vec<String> = args.arg_addons.split(",").map(|s| s.to_string()).collect();
-                addons.par_iter().for_each(|addon| {
-                    if args.flag_force {
-                        files::clear_pbo(&p, &addon).unwrap();
-                    }
-                    build::build_single(&p, &addon).print_error(true);
-                });
-            } else {
-                if args.flag_force {
-                    files::clear_pbos(&p).unwrap();
+            for addon in crate::files::all_addons() {
+                if skip.contains(&addon.file_name().unwrap().to_str().unwrap().to_owned()) { continue };
+                addons.push(addon);
+            }
+        }
+        if args.flag_force {
+            for addon in &addons {
+                if !skip.contains(&addon.file_name().unwrap().to_str().unwrap().to_owned()) {
+                    crate::files::clear_pbo(&p, &addon).unwrap_or_print();
                 }
-                build::build(&p).print_error(true);
             }
-            println!("  {} {}", "Finished".green().bold(), &p.name);
+        }
+        let mut state = crate::state::State::new(&addons);
+        p.run(&state).unwrap_or_print();
+        let result = build::many(&p, &addons).unwrap_or_print();
+        state.stage = crate::state::Stage::PostBuild;
+        state.result = Some(&result);
+        p.run(&state).unwrap_or_print();
+        if args.flag_release {
+            build::release::release(&p, &version).unwrap_or_print();
+            state.stage = crate::state::Stage::ReleaseBuild;
+            p.run(&state).unwrap_or_print();
+            println!("  {} {} v{}", match result.failed.len() {
+                0 => "Finished".green().bold(),
+                _ => "Finished".yellow().bold(),
```

Comment:
Should this be `error: Could not finish` (`error:` in red) like Cargo on compile error?

Reply:
I avoided red because it doesn't stop on failure. If we changed HEMTT to stop on failure red would be used, but I didn't feel a big `error` in red was suitable for a partial fail. That actual PBO that failed gets a red `error:`.

Reply:
True, yellow is fine. Maybe add a note that not all PBOs were successful?